<a href="https://colab.research.google.com/github/VPDeb/DS-Unit-1-Sprint-2-Statistics/blob/master/Debebe_LS_DS_123_Confidence_Intervals_Assignment_No_Answers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [ ]:
#Import statements
!wget https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv
import pandas as pd
import numpy as np
from scipy.stats import t
coca = pd.read_csv('Coke.csv')


--2020-09-09 20:37:23--  https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 492 [text/plain]
Saving to: ‘Coke.csv.2’

Coke.csv.2          100%[===================>]     492  --.-KB/s    in 0s      

2020-09-09 20:37:23 (27.5 MB/s) - ‘Coke.csv.2’ saved [492/492]



In [ ]:
#Print the top 5 rows
coca.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [ ]:
#Calculations here
mn_caf = coca['Caffeine'].mean()
sd_caf = coca['Caffeine'].std()
n_caf = coca['Caffeine'].count()
print(mn_caf)
print(sd_caf)
print(n_caf)

37.9402
5.243756828216712
50


In [ ]:
se_caf = 5.24/(50**(1/2))
se_caf

0.7410479066835018

The mean of the caffiene contect in Coke is 37.94
we can use this information along with the Standard Error which we get from dividing the the other parameters.

In [45]:
from scipy import stats 
def confidence_interval(data, confidence=0.95):
  """
  Calculate a confidence interval around a sample mean for given data.
  Using t-distribution and two-tailed test, default 95% confidence. 
  Arguments:
    data - iterable (list or numpy array) of sample observations
    confidence - level of confidence for the interval
  Returns:
    tuple of (mean, interval, lower bound, upper bound)
  """
  data = data[~np.isnan(data)]
  mean = np.mean(data)
  n = len(data)
  stderr = stats.sem(data)
  t = stats.t.ppf((1 + confidence) / 2.0, n - 1)
  interval = stderr * t
  return (mean, mean - interval, mean + interval);

Sentence or two summary.

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [ ]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=49)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [ ]:
#Calculations here
moe_caf = t_star*se_caf
moe_caf


1.4891915208412356

We are 95% confident that the range of caffeine in a 12oz Coke is between 36.45 and 39.43


###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [44]:
#Calculations here
coca.describe()

Drink   Caffeine
count  50.00000  50.000000
mean   25.50000  37.940200
std    14.57738   5.243757
min     1.00000  28.430000
25%    13.25000  33.662500
50%    25.50000  37.485000
75%    37.75000  41.737500
max    50.00000  48.120000

In [46]:
moe_caf2 = t_star*se_caf
moe_caf2

1.4891915208412356

In [47]:
lower = mn_caf - moe_caf2
lower

36.45100847915876

In [48]:
upper = mn_caf + moe_caf2
upper

39.429391520841236

###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [ ]:
#Calculations here
t.interval(alpha=0.95, df=49, loc=mn_caf, scale=se_caf)

(36.45100847915876, 39.429391520841236)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

Answer here.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

Interpretation here.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [ ]:
#Calculations here
t.interval(alpha=0.90, df=49, loc=mn_caf, scale=se_caf)

(36.69779547109857, 39.18260452890142)

A 90% confidence would allow for a smaller range in numbers and mean you believe you are being more precise in your computations.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [ ]:
#Calculations here
t.interval(alpha=0.99, df=49, loc=mn_caf, scale=se_caf)

Having a 99% confidence level has a larger range and therefore could be more accurate in including the correct value.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Answer here.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

Increasing the sample size would still change the margin of error due to the fact that the margin of error includes the t. computation as well as the standard error formula.  The margin of error would become smaller as we are testing a larger sample size.

In [ ]:
#Calculations here.
se_cafn = 5.24/(100**(1/2))
se_cafn

0.524

In [ ]:
t_starn = t.ppf(0.975,df=99)
t_starn

1.9842169515086827

In [ ]:
moe_cafn = t_starn*se_cafn
moe_cafn

1.0397296825905498

Compare MEs here.

